libs/lib-a/build.gradle.kts

```diff
@@ -21,15 +15,8 @@ dependencies {
   testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.8.2")
 }
 
-tasks.withType<Test> {
-  useJUnitPlatform()
-}
-
-val jar: Jar by tasks
-jar.apply {
-  archiveFileName.set("${project.name}-${project.version}.jar")
+tasks.named("jar", Jar::class.java) {
```

Comment:
Was a few less lines of code to do it this way. Version is done for all projects in the convention plugin.